xds/src/main/java/io/grpc/xds/LameFilter.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2021 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.xds;
+
+import com.google.common.util.concurrent.MoreExecutors;
+import com.google.protobuf.Message;
+import io.grpc.CallOptions;
+import io.grpc.Channel;
+import io.grpc.ClientCall;
+import io.grpc.ClientInterceptor;
+import io.grpc.Context;
+import io.grpc.LoadBalancer.PickSubchannelArgs;
+import io.grpc.Metadata;
+import io.grpc.MethodDescriptor;
+import io.grpc.Status;
+import io.grpc.xds.Filter.ClientInterceptorBuilder;
+import java.util.concurrent.Executor;
+import java.util.concurrent.ScheduledExecutorService;
+import javax.annotation.Nullable;
+
+/**
+ * A filter that fails all RPCs. To be added to the end of filter chain if RouterFilter is absent.
+ */
+enum LameFilter implements Filter, ClientInterceptorBuilder {
+  INSTANCE;
+
+  static final FilterConfig LAME_CONFIG = new FilterConfig() {
+    @Override
+    public String typeUrl() {
+      throw new UnsupportedOperationException();
+    }
+  };
+
+  @Override
+  public String[] typeUrls() {
+    return new String[0];
+  }
+
+  @Override
+  public StructOrError<? extends FilterConfig> parseFilterConfig(Message rawProtoMessage) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public StructOrError<? extends FilterConfig> parseFilterConfigOverride(Message rawProtoMessage) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Nullable
+  @Override
+  public ClientInterceptor buildClientInterceptor(
```

Comment:
I am very confused by how this method would be used. The `LameFilter` does not implement config parsing. That means you do not get a `FilterConfig`. In the xDS message parsing for top-level filters, if the config message doesn't exist, we just skip that filter entirely. That means, to use the `LameFilter`, we must have a config. Parsing the `LameFilter`'s config throws `UnsupportedOperationException`. I am deadlocked...

Reply:
LameFilter is not parsed from xDS message. It is added by grpc to the end of filter chain if RouterFilter is absent as in the spec :

```
if the router filter is not present, then gRPC will insert a special filter at the end of the filter chain that fails all RPCs. 
```

`parseFilterConfig()` will never be called.

There are many ways to do it equivalently. I chose a way that is closest to c-core.